### ISSUE TITLE
composing distribution names

### ISSUE THREAD

**Issue by e2thenegpii** [pypa/packaging#408]
Is there interest in a pull request for functions that perform the inverse operation of ```packaging.utils.parse_wheel_filename``` and ```parse_sdist_filename```?  

It seems like an omission that there is no function such that ```x == compose_wheel_filename(*parse_wheel_filename(x))```

**Comment by brettcannon** [comment]
Not so much an omission as no one has shown a need yet and we prefer to only add support for things people have demonstrable need for instead of guessing. If the wheel project told us they could use such a thing, though, we would consider adding it.

**Comment by uranusjr** [comment]
Also other PEP 517 backend authors since they would have need for it.

**Comment by MrMino** [comment]
I have somewhat of a need for it in my project, so I can PR this if you're willing to accept it.

**Comment by brettcannon** [comment]
@MrMino can I ask why it's only "somewhat"? If you have an actual need then we will totally look at a PR, but I would rather avoid speculative code.

**Comment by pradyunsg** [comment]
Isn't it going to be just "-".join(...) + extension?

Does this really need to be a function here?

**Comment by MrMino** [comment]
@brettcannon Well, I'm doing it inside [wheelfile](https://github.com/MrMino/wheelfile/blob/master/wheelfile.py#L1050). It isn't a big deal at all, I just feel like it would be elegant to have it. I haven't utilized `parse_wheel_filename` there yet, but I'm planning on doing it.

@pradyunsg technically `parse_wheel_filename` returns `Version` and a set of tags, so you have to cast it properly, then there's also the kink in handling the build number.

**Comment by brettcannon** [comment]
It's a little more complicated than that due to tag sets. If we added a function to construct that part of the wheel file name then you might as well just skip having that as a separate function (since I don't think it would be _that_ useful on its own) and go as far as constructing the wheel file name entirely.

As an example, this is untested code written for Python 3.9:
```Python
import operator

def _compress_wheel_tags(tags: AbstractSet[Tag], attr: str) -> str:
    tag_portions = []
    for attr in "interpreter", "abi", "platform":
        parts = map(operator.attrgetter(attr), tags)
        compressed = ".".join(sorted(parts))
        tag_portions.append(compressed)
    return "-".join(tag_portions)


def create_wheel_filename(project: NormalizedName, version: Version, build: BuildTag | None, tags: AbstractSet[Tag]) -> str:
    compressed_tag = _compress_wheel_tags(tags)
    if build:
        parts = project, version, build, compressed_tag
    else:
        parts = project, version, compressed_tag
    return "-".join(map(str, parts)) + ".whl"
```

**Comment by uranusjr** [comment]
If I were to ask for such a function (I am not), I would want name and version normalisation, ensuring components don’t contain `-`, things like that.

**Comment by brettcannon** [comment]
@MrMino fair enough! If you would use it if we provided it then I'm okay with adding the function.

@uranusjr I would flat-out dictate via typing that stuff had to be normalized.

**Comment by brettcannon** [comment]
FYI there is a PR for this, but I think we need to resolve [PEP 625](https://peps.python.org/pep-0625/) first.

**Comment by brettcannon** [comment]
FYI https://peps.python.org/pep-0625/ has been accepted!

**Comment by e2thenegpii** [comment]
I've resoled the merge conflicts in the PR.
